### ISSUE TITLE
Displaying help documention

### ISSUE THREAD

**Issue by Geraint87** [adamabdelhamed/PowerArgs#30]
How do you set up the help for a parser that uses the Action Framework?

If I paraphrase the superencoder code snippets, what I'd want/expect is something does:

```
superencoder -help
```

-- displays the help docs

```
superencoder encode -help
```

-- displays EncodeArgs help docs

```
superencoder clip -help
```

-- displays ClipArgs help docs

At the moment, what I actually get is the following behaviour:

```
superencoder -help
```

-- displays the help docs **and the error "No action was specified"**

```
superencoder encode -help
```

-- displays EncodeArgs help docs **and the error "The argument 'Source' is required"**

```
superencoder clip -help
```

-- displays ClipArgs help docs **and the error "The argument 'Source' is required"**

So basically:
1. Can you add a -help property to a parent class (which has actions) without having to specify an Action?
2. Is it possible to make arguments Required when the user _tries_ to use them, but not, for example, if another property, bool Help = true?

Cheers!
